hardware/megacli.py

```diff
@@ -42,9 +42,8 @@ def which(cmd, mode=os.F_OK | os.X_OK, path=None):
     # Check that a given file can be accessed with the correct mode.
     # Additionally check that `file` is not a directory, as on Windows
     # directories pass the os.access check.
-    def _access_check(fn, mode):
-        return (os.path.exists(fn) and os.access(fn, mode)
-                and not os.path.isdir(fn))
+    def _access_check(myfile, mode):
+        return os.path.exists(myfile and os.access(myfile, mode) and not os.path.isdir(myfile))
```

Comment:
This is broken:- TypeError: coercing to Unicode: need string or buffer, bool found\n'

example log:- https://logs.rdoproject.org/43/13943/83/check/periodic-tripleo-ci-centos-7-ovb-3ctlr_1comp-featureset035-master/568d735/logs/undercloud/var/log/containers/ironic-inspector/ironic-inspector.log.txt.gz#_2019-02-27_05_50_07_373

Reply:
I'm really sorry for that one.
Pushing the fix.

Reply:
Fixed by https://github.com/redhat-cip/hardware/commit/f9f94bec9d24d074e13f8ee0dd35eb5834f31a32